apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -278,6 +278,28 @@ type RAIDConfig struct {
 	SoftwareRAIDVolumes []SoftwareRAIDVolume `json:"softwareRAIDVolumes,omitempty"`
 }
 
+// FirmwareConfig contains the configuration that you want to configure BIOS settings in Bare metal server
+type FirmwareConfig struct {
+	// Whether reset firmware settings.
+	// NOTE: IRMC does not support reset firmware settings.
+	ResetSettings bool `json:"resetSettings,omitempty"`
```

Comment:
This option doesn't seem very declarative, and it's not discussed in the [design doc](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/bios-config.md). I'm not a fan of it.